### ISSUE TITLE
SGDR: Stochastic Gradient Descent with Warm Restarts

### ISSUE THREAD

**Issue by c-bata** [c-data/survey#6]
arXiv: https://arxiv.org/abs/1608.03983
GitHub: https://github.com/loshchil/SGDR
conference: ICLR 2017

SGDのsimpleなwarm restart手法を提案。PyTorchに実装されている [CosineAnnealingLR](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.CosineAnnealingLR) はこの論文で提案されたらしい。

lossやgradientsはデータのばっちごとに大きく異なる可能性があるため、incoming informationのノイズを除去する必要がある。各バッチのcosine annealingで学習率を減衰させていく。

![スクリーンショット 2019-12-10 14 30 07](https://user-images.githubusercontent.com/5564044/70498404-97244f80-1b59-11ea-92a5-66976395aae8.png)
